genesis-tools/contracts/lockup-contract/src/lib.rs

```diff
@@ -0,0 +1,96 @@
+//! A smart contract that allows tokens lockup.
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use near_bindgen::{env, near_bindgen, Promise};
+
+#[global_allocator]
+static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
+
+/// A key is a sequence of bytes, potentially including the prefix determining the cryptographic type
+/// of the key. For forward compatibility we do not enforce any specific length.
+type PublicKey = Vec<u8>;
+
+#[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct LockupContract {
+    /// The amount in yacto-NEAR tokens locked for this account.
+    lockup_amount: u128,
+    /// The timestamp in nanoseconds when the lockup amount of tokens will be available.
+    lockup_timestamp: u64,
+}
+
+impl Default for LockupContract {
+    fn default() -> Self {
+        env::panic(b"The contract is not initialized.");
```

Comment:
when does this gets called? This seems like important piece that we should have by default if there is `[init]`

Reply:
It was the logic before, but with multiple `[init]`'s we can't do this automatically.